### PR TITLE
[Cert-manager]: Sync upstream master and add cert-manager relationships

### PR DESCRIPTION
**Pull request by innocentrda**
**Notes for Reviewers**

This PR fixes #

Screenshot from 2024-05-16 13-38-36

In cert-manager, a secret may contain certificate data, and the Issuer resource references the secret resource by name to access the certificates. In other words, the Issuer is  is patched with the configuration of the Secret. This is Hierarchical-Inventory relationship similar to Secret-Pod relationship. 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by deepak4566 — CHANGES_REQUESTED** [review]
A proof of working relationship as video or screenshot would be benificial

File: server/meshmodel/cert-manager/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  "kind": "Hierarchical",
+  "metadata": {
+    "description": "A hierarchical inventory relationship in which the configuration of (parent) component is patched with the configuration of other (child) component. E.g Secret->Pod, ConfigMaps->Deployment, etc..."
+  },
+  "model": {
+    "schemaVersion": "core.meshery.io/v1beta2",
+    "version": "v1.0.0",
+    "name": "cert-manager",
+    "model": {
+      "version": "1.14"
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply server/meshmodel/cert-manager/relationships/hierarchical_inventory.json:13]: @innocentrda yes/no?

[the remark this answers is not in the document]
Comment (leecalcote) [reply server/meshmodel/cert-manager/relationships/hierarchical_inventory.json:13]: Please add this as a Relationship authoring best practice.

[the remark this answers is not in the document]
Comment (innocentrda) [reply server/meshmodel/cert-manager/relationships/hierarchical_inventory.json]: Yes it's done

**Review by leecalcote — APPROVED** [review]

**Pull request merged**